File: routes/vehicles.py
from flask import Blueprint, request, jsonify
from models import db, Vehicle

vehicles_blueprint = Blueprint("vehicles", __name__)

@vehicles_blueprint.route("/", methods=["GET"])
def get_all_vehicles():
    vehicles = Vehicle.query.all()
    result = [
        {
            "id": v.id,
            "owner_id": v.owner_id,
            "type": v.type,
            "location": v.location,
            "availability": v.availability,
        }
        for v in vehicles
    ]
    return jsonify(result), 200

@vehicles_blueprint.route("/", methods=["POST"])
def add_vehicle():
    data = request.json
    new_vehicle = Vehicle(
        owner_id=data["owner_id"],
        type=data["type"],
        location=data["location"],
        availability=True,
    )
    db.session.add(new_vehicle)
    db.session.commit()
    return jsonify({"message": "Vehicle added successfully"}), 201

File: routes/vehicles.js
const express = require("express");
const Vehicle = require("../models/Vehicle");
const router = express.Router();

// 모든 차량 가져오기
router.get("/", async (req, res) => {
  try {
    const vehicles = await Vehicle.find();
    res.json(vehicles);
  } catch (err) {
    res.status(500).json({ message: "Server Error" });
  }
});

// 차량 추가
router.post("/", async (req, res) => {
  const { ownerId, type, location } = req.body;

  try {
    const newVehicle = new Vehicle({ ownerId, type, location });
    await newVehicle.save();
    res.status(201).json({ message: "Vehicle added successfully" });
  } catch (err) {
    res.status(500).json({ message: "Failed to add vehicle" });
  }
});

module.exports = router;
